apps/src/publicKeyCryptography/ModuloClock.jsx

```diff
@@ -1,19 +1,188 @@
 /** @file Centered animated Modulo Clock component for the Crypto widget levels */
+import _ from 'lodash';
 import React from 'react';
 import color from '../color';
 
-const MODULO_CLOCK_RADIUS = 150;
+// Defines the coordinate scale for SVG elements
+const VIEWBOX_SIDE = 100;
+// Above this size, the slices of the clock will have less space between them
+const SMALL_GAPS_OVER_MODULUS = 200;
+// Above this size, the clock will not render individual slices
+const CONTINUOUS_METER_OVER_MODULUS = 400;
+// Radius of the outer edge of a wedge clock segment
+const WEDGE_OUTER_RADIUS = 48;
+// Radius of the inner edge of a wedge clock segment
+const WEDGE_INNER_RADIUS = 25;
+
+const COLOR = {
+  clockFace: color.lightest_cyan,
+  emptyWedge: color.white,
+  fullWedge: color.cyan,
+  valueText: color.cyan
+};
+
 const style = {
-  width: 2 * MODULO_CLOCK_RADIUS,
-  height: 2 * MODULO_CLOCK_RADIUS,
-  margin: 'auto',
-  borderRadius: MODULO_CLOCK_RADIUS,
-  backgroundColor: color.teal
+  root: {
+    textAlign: 'center',
+    marginTop: 30,
+    marginBottom: 30
+  },
+  svg: {
+    width: 300,
+    height: 300
+  }
 };
 
-export default function ModuloClock() {
-  return (
-    <div>
-      <div style={style}/>
-    </div>);
+const ModuloClock = React.createClass({
+  propTypes: {
+    modulus: React.PropTypes.number
+  },
+
+  getInitialState() {
+    return {
+      currentDividend: 0,
+      targetDividend: 0,
+      startTime: null,
+      interval: null,
+      onStep: null,
+      onComplete: null
+    };
+  },
+
+  /**
+   * @param {number} dividend - The left-hand-side of the modulus operation
+   * @param {number} speed - A value from 1 to 10 where 10 is the fastest option
+   * @param {function} onStep - callback called on each frame of animation,
+   *        which lets us sync other work to it.
+   * @param {function} onComplete - callback called at end of animation
+   */
+  animateTo(dividend, speed, onStep, onComplete) {
+    const MAXIMUM_TIME_PER_SEGMENT = 500 / speed;
+    const MAXIMUM_TOTAL_TIME = 8000 / speed + 2000;
+    this.setState({
+      currentDividend: 0,
+      targetDividend: dividend,
+      startTime: Date.now(),
+      duration: Math.min(MAXIMUM_TOTAL_TIME, dividend * MAXIMUM_TIME_PER_SEGMENT),
+      interval: setInterval(this.tick, 33),
+      onStep: onStep || function () {},
+      onComplete: onComplete || function () {}
```

Comment:
I wonder if some of these would be better served as instance variables rather than state. i.e. When I change interval, do I need to render? (I think "no", because presumably the `tick` method will subsequently change state.